ENIGMAsystem/SHELL/Platforms/xlib/LINUXjoystick.cpp

```diff
@@ -159,6 +166,8 @@ namespace enigma_user
     printf("Joystick name: %s\n",name);
     #endif
 
+    // There is no guarantee that because /dev/js5 exists, /dev/js0 exists
+    if (enigma::joysticks.size() <= id) enigma::joysticks.resize(id);
```

Comment:
I think this is indicative of a fundamentally bad decision made by this joystick code. I'm not sure, but I think I disagree with having gaps in joystick ids. I believe one reason Overmar's version only has 2 joysticks is because he enumerates all of them with the Win32 API and just ignores joysticks that are unusable and not connected. I don't think that's a good reason to limit the number of joysticks, I just don't know how great of an idea it is to expect the user to search for the joystick id.